From 74499a829c26436357fcd75e0e869a0cbdc5a9ce Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 21 Apr 2019 16:09:03 +0300
Subject: [PATCH] Revert "[SW-0] pll: Gaudi skip init on warm reset"

There should be no need to skip all PLLs initialization, only a few.
The problematic PLLs have been found and a follow up patch is coming.
Revert the current W/A.

This reverts commit 052612c90671ce81bf2b56c705a8b3c075d4a7de.

Change-Id: Icfc262e40fecda702f91f7820f8e0ca53749c3b3
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index d4cb726..71e3da8 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -191,7 +191,7 @@
 /**
  * hl_pll_init - Initialize PLL registers
  */
-int hl_pll_init(unsigned int *pll_partial_init)
+void hl_pll_init(void)
 {
 	int i, pll_num;
 	struct hl_pll_cfg *hl_pll_cfg;
@@ -203,20 +203,7 @@
 	hl_pll_udelay_init();
 
 	pll_num = hl_pll_get_cfg(&hl_pll_cfg);
-	if (!pll_num)
-		return 1;
 
-	*pll_partial_init = hl_pll_partial_init_get();
-
-	for (i = 0; i < pll_num; i++) {
-		if (*pll_partial_init)
-			hl_pll_bypass(hl_pll_cfg + i, false);
-		else
-			hl_pll_set(hl_pll_cfg + i);
-	}
-
-	if (!*pll_partial_init)
-		hl_pll_partial_init_set();
-
-	return 0;
+	for (i = 0; i < pll_num; i++)
+		hl_pll_set(hl_pll_cfg + i);
 }
diff --git a/common/pll.h b/common/pll.h
index 8845630..f1ab877 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -59,12 +59,9 @@
 
 enum hl_pll_bypass_val hl_pll_bypass_get(u64 pll_base,
 					 enum hl_pll_div_out out);
-int hl_pll_init(unsigned int *pll_partial_init);
+void hl_pll_init(void);
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg);
 u64 hl_pll_get_cpu_cycles_per_usec(void);
 u32 hl_pll_get_freq(u64 pll_base, enum hl_pll_div_out div);
-
-int hl_pll_partial_init_get(void);
-void hl_pll_partial_init_set(void);
 
 #endif /* _HL_PLL_H_ */
diff --git a/common/preboot.c b/common/preboot.c
index d9cd96d..89654d0 100644
--- a/common/preboot.c
+++ b/common/preboot.c
@@ -152,9 +152,7 @@
 
 void preboot_main(void)
 {
-	int ret;
 	void (*spl_start)(void) = (void *)SPL_TEXT_BASE;
-	unsigned int pll_partial_init = 0;
 
 	regmap_writel(CPU_BOOT_STATUS_IN_PREBOOT, get_cpu_boot_status_reg());
 
@@ -163,20 +161,9 @@
 	preboot_pcie_init();
 	preboot_delay();
 	preboot_sync_btl();
-	ret = hl_pll_init(&pll_partial_init);
+	hl_pll_init();
 	/* reinit uart after PLLs configured */
 	hl_uart_init();
-
-	/* Now that UART is ready notify about PLL init */
-	if (ret)
-		preboot_puts("Skip PLLs setup [DIP switch is set]!\n");
-	else {
-		if (pll_partial_init)
-			preboot_puts("Partial PLLs setup [warm reset]!\n");
-		else
-			preboot_puts("Full PLLs setup [cold reset]!\n");
-	}
-
 	preboot_copy_spl();
 
 	spl_start();
diff --git a/gaudi/common.c b/gaudi/common.c
index 90d960a..d3b289d 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -43,13 +43,3 @@
 {
 	return mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_3;
 }
-
-int hl_pll_partial_init_get(void)
-{
-	return regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
-}
-
-void hl_pll_partial_init_set(void)
-{
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
-}
diff --git a/goya/common.c b/goya/common.c
index f73b49d..0dce834 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -170,10 +170,3 @@
 {
 	return mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3;
 }
-
-int hl_pll_partial_init_get(void)
-{
-	return 0;
-}
-
-void hl_pll_partial_init_set(void) {}
